jpaNew/src/main/java/com/example/jpaNew/repositories/SocialMediaPostsRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.example.jpaNew.repositories;
+
+
+import com.example.jpaNew.entities.SocialMediaPosts;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface SocialMediaPostsRepository extends JpaRepository<SocialMediaPosts, Long> {
+    List<SocialMediaPosts> findByIdGreaterThanEqualAndIdLessThanEqual(Long startId, Long endId);
```

Comment:
Here you have made it list and which is correct ! How come same though did not cross in your mind!
I want to smoke what you are smoking!

Reply:
make all methods of interface public